gcampus/analysis/views.py

```diff
@@ -1,11 +1,32 @@
+import math
+
 from django.utils.translation import gettext_lazy as _
-from django.views.generic import ListView
+from django.views.generic import ListView, DetailView
 
-from gcampus.core.models import ParameterType
+from gcampus.core.models import ParameterType, Calibration
 from gcampus.core.views.base import TitleMixin
 
 
-class ODConverterView(TitleMixin, ListView):
+class ODConverterOverView(TitleMixin, ListView):
     title = _("Optical Density Converter")
-    template_name = "gcampusanalysis/od_converter.html"
+    template_name = "gcampusanalysis/od_converter_overview.html"
     model = ParameterType
+    context_object_name = "parameter"
+
+    def get_context_data(self, **kwargs):
+        kwargs["row_num"] = math.ceil(len(ParameterType.objects.all()) / 3)
+        kwargs["param_num"] = len(ParameterType.objects.all())
+        return super(ODConverterOverView, self).get_context_data(**kwargs)
+
+
+class ODConverterDetailView(TitleMixin, DetailView):
+    title = _("Optical Density Converter")
+    template_name = "gcampusanalysis/od_converter_detail.html"
+    model = ParameterType
+    context_object_name = "parameter"
+
+    def get_context_data(self, **kwargs):
+        kwargs["calibrations"] = Calibration.objects.all().filter(
```

Comment:
While this has virtually no effect, normally you call `filter(...)` before `all()`. When you take a look at the code of `all()`, you can see that all it does is to return a copy of the queryset. In many places it can be omitted, but it is best to just put it at the end of a query if unsure.

Reply:
fixed in 72a9f26